src/react-components/ui-root.js

```diff
@@ -195,6 +204,37 @@ class UIRoot extends Component {
     this.props.scene.addEventListener("share_video_disabled", this.onShareVideoDisabled);
     this.props.scene.addEventListener("exit", this.exit);
     const scene = this.props.scene;
+
+    let preEntryHistoryLength = 0;
+
+    // If we landed on the page with a path in the middle of the entry flow (eg we refreshed the
+    // page on the audio setup dialog) then reset the history entry to /.
+    //
+    // Note this isn't perfect, if we refresh the page mid-entry flow and then hit back, we end
+    // up in a bad state unless we were on the first step. But this seems reasonable enough for now.
+    if (ENTRY_FLOW_ROUTES.find(x => x === this.props.history.location.pathname)) {
```

Comment:
Not 100% sure but I think you can push states with the same path, so instead of making each entry flow step its own path we can just have a pushState that has the current flow step in it, that way back button still works, but the url doesn't change so refreshing still goes back to the beginning for the flow. This combined with a replace on the actual entry should fix these 2 hacks.